x/provider/query/types.go

```diff
@@ -44,3 +45,13 @@ func (obj Providers) String() string {
 func (p *Provider) Address() sdk.AccAddress {
 	return p.Owner
 }
+
+// MatchReqAttributes method compares provided attributes with specific provider require attributes
+func (p Provider) MatchReqAttributes(attrs []sdk.Attribute) bool {
```

Comment:
Should it be renamed to `MatchRequiredAttributes` ? `Req` is usually shortcut for `request`